crates/tree-sitter-perl-c/c-src/scanner.c

```
@@ -0,0 +1,988 @@
+  if (c == '}') ADVANCE_C;
+}
+
+static bool isidfirst(int32_t c) { return c == '_' || is_tsp_id_start(c); }
+
+static bool isidcont(int32_t c) { return c == '_' || is_tsp_id_continue(c); }
+
+// in any interpolatable case, we wanna stop parsing on these chars
+// there's a matching rule in the grammar to catch when it doesn't match a rule
+static bool is_interpolation_escape(int32_t c) { return c < 256 && tsp_strchr("$@-[{\\", c); }
+
+unsigned int tree_sitter_perl_external_scanner_serialize(void *payload, char *buffer) {
```

Cppcheck (2.20.0)

[style] The function 'tree_sitter_perl_external_scanner_serialize' is never used.

(unusedFunction)